src/autobridgecalculator.js

```diff
@@ -70,14 +73,17 @@ AutobridgeCalculator.prototype.calculate = function() {
       continue;
     }
 
-    if (legOneTakerGetsFunded.compareTo(legTwoTakerPaysFunded) > 0) {
+    //  using private fields for speed
```

Comment:
Maybe you could add a guard statement to parse_json to return fast if the argument is already an Amount? And maybe there is a way to speed up Amount creation in the Amount class?